### ISSUE TITLE
Error with 'amplify upgrade': Cannot find module '/Users/hisham/src/ess-app/--version'

### ISSUE THREAD

**Issue by hisham** [aws-amplify/amplify-cli#10583]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.17.6

### Amplify CLI Version

8.5.0

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No

### Amplify Categories

Not applicable

### Amplify Commands

Not applicable

### Describe the bug

I ran amplify upgrade command got following error: 

`hisham@mb3 ess-app % amplify upgrade

   ╭────────────────────────────────────────────────────────────╮
   │                                                            │
   │                     Update available:                      │
   │   Run amplify upgrade for the latest features and fixes!   │
   │                                                            │
   ╰────────────────────────────────────────────────────────────╯

Downloading latest Amplify CLI
100% [========================================] 0.0 seconds left
⠧ Validating downloaded Amplify CLI....internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module '/Users/hisham/src/ess-app/--version'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function._resolveFilename (pkg/prelude/bootstrap.js:1819:46)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Function.runMain (pkg/prelude/bootstrap.js:1847:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
🛑 Command failed with exit code 1: /Users/hisham/.amplify/bin/amplify --version
Error: Command failed with exit code 1: /Users/hisham/.amplify/bin/amplify --version
    at makeError (/snapshot/repo/build/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/snapshot/repo/build/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async upgradeCli (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/commands/upgrade.js:100:24)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/commands/upgrade.js:57:9)
    at async Object.executeAmplifyCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:285:9)
    at async executePluginModuleCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:142:5)
    at async executeCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:40:9)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:163:9)`

### Expected behavior

amplify upgrade should work

### Reproduction steps

Not sure as amplify did upgrade, but I guess the upgrade validation step failed. Running amplify upgrade again tells me I have latest cli. 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @hisham :wave: thanks for raising this! I've marked this as a bug 🙂 this appears to be a regression of https://github.com/aws-amplify/amplify-cli/issues/10528, however I will let the team classify this further

**Comment by sachscode** [comment]
Hi @hisham 
There was a bug in `amplify upgrade` in amplify-cli 8.4.0 which has been fixed in amplify-cli 8.5.0 . Please install the CLI using `npm install -g @aws-amplify/cli@8.5.0` . The problem in amplify upgrade will not be seen in future versions of the CLI. Closing this, since resolved in 8.5.0.

**Comment by elankeeran** [comment]
I updated to `npm install -g @aws-amplify/cli@8.5.0` version but still Im getting warn `Overrides functionality is not implemented for this category`
